### PR TITLE
Replace `RequestContext` kwarg `tenant` with `project_id`

### PR DESCRIPTION
**Pull request by Dany9966**

### CODE REVIEW

**Review by aznashwan — CHANGES_REQUESTED** [review]
Would rather avoid legacy `tenant` attribute altogether.

File: coriolis/context.py [hunk 1]
```diff
@@ -37,6 +37,7 @@ def __init__(self, user, tenant, is_admin=None,
                                              show_deleted=show_deleted,
                                              request_id=request_id,
                                              overwrite=overwrite)
+        self.tenant = tenant
```
[the remark this answers is not in the document]
Comment (Dany9966) [reply coriolis/context.py:40]: Yes, it is accessed in a bunch of places, most commonly by the DB API when getting objects by project_id

**Comment by aznashwan** [comment]
LGTM. I'd like to run this through proper testing though. (most notably testing minion pools and replica schedules, since they directly depend on the trust setup)

**Comment by Dany9966** [comment]
Minion pools and replica schedules have been double-checked.

**Pull request merged**